grr/server/grr_response_server/databases/spanner_cron_jobs.py

```diff
@@ -0,0 +1,724 @@
+        "RunId",
+        "CreationTime",
+        "FinishTime",
+        "Status",
+        "LogMessage",
+        "Backtrace",
+    ]
+    rowrange = spanner_lib.KeyRange(start_closed=[job_id], end_closed=[job_id])
+    rows = spanner_lib.KeySet(ranges=[rowrange])
+
+    res = []
+    for row in self.db.ReadSet(table="CronJobRuns",
+                               rows=rows,
+                               cols=cols,
+                               txn_tag="ReadCronJobRuns"):
+      res.append(
+          _CronJobRunFromRow(
+              job_run=row[0],
+              job_id=row[1],
+              run_id=row[2],
+              creation_time=row[3],
+              finish_time=row[4],
+              status=row[5],
+              log_message=row[6],
+              backtrace=row[7],
+          )
+      )
+
+    return sorted(res, key=lambda run: run.started_at or 0, reverse=True)
+
+  @db_utils.CallLogged
+  @db_utils.CallAccounted
+  def ReadCronJobRun(self, job_id: str, run_id: str) -> flows_pb2.CronJobRun:
+    """Reads a single cron job run from the db.
+
+    Args:
+      job_id: The job_id of the run to be read.
+      run_id: The run_id of the run to be read.
+
+    Returns:
+      An flows_pb2.CronJobRun object.
+    """
+    cols = [
+        "Payload",
+        "JobId",
+        "RunId",
+        "CreationTime",
+        "FinishTime",
+        "Status",
+        "LogMessage",
+        "Backtrace",
+    ]
+    try:
+      row = self.db.Read(table="CronJobRuns",
+                         key=(job_id, run_id),
+                         cols=cols,
+                         txn_tag="ReadCronJobRun")
+    except NotFound as error:
+      raise db.UnknownCronJobRunError(
+          "Run with job id %s and run id %s not found." % (job_id, run_id)
+      ) from error
+
+    return _CronJobRunFromRow(
+        job_run=row[0],
+        job_id=row[1],
+        run_id=row[2],
+        creation_time=row[3],
+        finish_time=row[4],
+        status=row[5],
+        log_message=row[6],
+        backtrace=row[7],
+    )
+
+  @db_utils.CallLogged
+  @db_utils.CallAccounted
+  def DeleteOldCronJobRuns(self, cutoff_timestamp: rdfvalue.RDFDatetime) -> int:
+    """Deletes cron job runs that are older than cutoff_timestamp.
+
+    Args:
+      cutoff_timestamp: This method deletes all runs that were started before
+        cutoff_timestamp.
+
+    Returns:
+      The number of deleted runs.
+    """
+    query = """
+    SELECT cjr.JobId, cjr.RunId
+      FROM CronJobRuns AS cjr
+     WHERE cjr.CreationTime < @cutoff_timestamp
+    """
+    params = {"cutoff_timestamp": cutoff_timestamp.AsDatetime()}
+
+    def Transaction(txn) -> int:
+      rows = list(txn.execute_sql(sql=query, params=params))
+
+      for job_id, run_id in rows:
+        keyset = spanner_lib.KeySet(keys=[[job_id, run_id]])
+        
+        txn.delete(table="CronJobRuns", keyset=keyset,
+                   request_options={"request_tag": "DeleteOldCronJobRuns:CronJobRuns:delete"})
```

Comment:
```suggestion
        txn.delete(
          table="CronJobRuns",
          keyset=keyset,
        )
```
`request_options` is not an argument to `delete()`.